external/kinesis-asl/src/main/scala/org/apache/spark/streaming/kinesis/KinesisDStream.scala

```diff
@@ -0,0 +1,559 @@
+     * @param stsSessionName Name to identify STS sessions by
+     * @param awsAccessKeyId AWS access key ID for authenicating to STS
+     * @param awsSecretKey AWS secret key for authenticating to STS
+     * @return Reference to this [[KinesisDStream.Builder]]
+     */
+    def cloudWatchSTSCredentials(
+        stsRoleArn: String,
+        stsSessionName: String,
+        awsAccessKeyId: String,
+        awsSecretKey: String): Builder[T] = {
+      cloudWatchCredsProvider = Option(buildSTSCredentials(
+        stsRoleArn,
+        stsSessionName,
+        awsKeypair = Option(awsAccessKeyId, awsSecretKey)))
+      this
+    }
+
+    /**
+     * Assume an STS role in order to authenticate to the CloudWatch service. The default
+     * credential chain is used to authenticate to STS and the provided external ID is
+     * validated against the value in the assumed role's policy (if one is specified).
+     *
+     * @param stsRoleArn ARN of the IAM role to assume via STS
+     * @param stsSessionName Name to identify STS sessions by
+     * @param stsExternaId External ID used by STS to validate IAM role trust policy
+     * @return Reference to this [[KinesisDStream.Builder]]
+     */
+    def cloudWatchSTSCredentials(
+        stsRoleArn: String,
+        stsSessionName: String,
+        stsExternalId: String): Builder[T] = {
+      cloudWatchCredsProvider = Option(buildSTSCredentials(
+        stsRoleArn,
+        stsSessionName,
+        stsExternalId = Option(stsExternalId)))
+      this
+    }
+
+    /**
+     * Assume an STS role in order to authenticate to the CloudWatch service. The provided
+     * AWS credential keypair is used to authenticate to STS and the provided external ID
+     * is validated against the value in the assumed role's policy (if one is specified).
+     *
+     * @param stsRoleArn ARN of the IAM role to assume via STS
+     * @param stsSessionName Name to identify STS sessions by
+     * @param awsAccessKeyId AWS access key ID for authenicating to STS
+     * @param stsExternaId External ID used by STS to validate IAM role trust policy
+     * @param awsSecretKey AWS secret key for authenticating to STS
+     * @return Reference to this [[KinesisDStream.Builder]]
+     */
+    def cloudWatchSTSCredentials(
+        stsRoleArn: String,
+        stsSessionName: String,
+        stsExternalId: String,
+        awsAccessKeyId: String,
+        awsSecretKey: String): Builder[T] = {
+      cloudWatchCredsProvider = Option(buildSTSCredentials(
+        stsRoleArn,
+        stsSessionName,
+        stsExternalId = Option(stsExternalId),
+        awsKeypair = Option(awsAccessKeyId, awsSecretKey)))
+      this
+    }
+
+    /**
+     * Create a new instance of [[KinesisInputDStream]] with configured parameters.
+     *
+     * @return Instance of [[KinesisInputDStream]] constructed with configured parameters
+     */
+    def build(): KinesisInputDStream[T] = new KinesisInputDStream(
```

Comment:
I'll expand the builder to recognize required params and remove them as ctor args